.github/workflows/docs.yml

```diff
@@ -21,7 +21,7 @@ jobs:
     - name: Install dependencies
       run: |
         python3 -m pip install --upgrade pip
-        python3 -m pip install -e '.[docs]'
+        python3 -m pip install -e '.[docs,pulse]'
```

Comment:
pulse extra shouldn't be required to build docs. openpulse should instead just be added to [autodoc_mock_imports](https://github.com/qBraid/pyqasm/blob/da60f1c7a33ccf97cdd3f9cfe4a26e04c3aaee28/docs/conf.py#L44)